falcon/routing/compiled.py

```diff
@@ -162,10 +174,11 @@ def line(text, indent_offset=0):
                 resource_idx = len(self._return_values)
                 self._return_values.append(node)
 
-            self._compile_tree(node.children, indent, level + 1)
+            self._compile_tree(node.children, indent, level + 1, fast_return)
 
             if node.resource is None:
-                line('return None')
+                if fast_return:
+                    line('return None')
```

Comment:
One thing that isn't overly clear to me is why we're adding these lines only if fast-return is True?  What happens If fast-return is false, such as the case where there are variable nodes?

(this is for all the `line('return None')` lines)